postgres/lib.w

```diff
@@ -32,22 +36,80 @@ pub struct DatabaseProps {
 
 pub interface IDatabase {
   inflight query(query: str): Array<Map<Json>>;
-  inflight connectionOptions(): ConnectionOptions;
+  inflight connectionOptions(): ConnectionOptions?;
+}
+
+struct RequiredTFAwsProps {
+  /**
+  * Supported: [neon, rds]
+  */
+  postgresEngine: str;
+}
+
+struct RequiredRDSParameters {
+  instanceCount: num;
+  /**
+   * See: https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.DBInstanceClass.html 
+   */
+  instanceClass: str;
+  /** Deault postgres */
+  masterUsername: str?;
+}
+
+struct AWSDatabaseRefParameters {
+  vpcId: str;
+}
+
+pub class DatabaseRef {
+  connectionSecret: cloud.Secret;
+
+  new() {
+    // For database refs, we just need to have a secret connection string to use for queries
+    this.connectionSecret = new cloud.Secret(name: "connectionString_{this.node.id}");
+
+    let target = util.env("WING_TARGET");
+    if target == "tf-aws" {
+      let params = nodeof(this).app.parameters;
+
+      let allParams = params.read();
+      if !allParams.has("tf-aws") {
+        // TODO: find a good way to enforce this with parameters. (currently cant duplicate required params schemas)
+        log("WARNING: Unless your database is accessible from the public internet, you must provide vpc info under `tf-aws` in your wing.toml file\nFor more info see: https://www.winglang.io/docs/platforms/tf-aws#parameters");
+      }
+    }
+  }
+
+  pub inflight query(query: str): Array<Map<Json>> {
+    return PgUtil._queryWithConnectionString(query, this.connectionSecret?.value()!);
```

Comment:
Does this work?
```suggestion
    return PgUtil._queryWithConnectionString(query, this.connectionSecret.value());
```

Reply:
Yupp, it works, must have been leftovers from something I was trying